File: infra/garage/configure-cors.js
import { S3Client, PutBucketCorsCommand } from "@aws-sdk/client-s3";
import dotenv from "dotenv";

dotenv.config({ quiet: true });

const { S3_ENDPOINT, S3_ACCESS_KEY, S3_SECRET_KEY, S3_BUCKET } = process.env;

const client = new S3Client({
  endpoint: S3_ENDPOINT,
  region: "garage",
  credentials: {
    accessKeyId: S3_ACCESS_KEY,
    secretAccessKey: S3_SECRET_KEY,
  },
  // Importante: O Garage geralmente não suporta caminhos virtuais (bucket.dominio)
  // forcePathStyle garante o formato dominio/bucket
  forcePathStyle: true,
});

const command = new PutBucketCorsCommand({
  Bucket: S3_BUCKET,
  CORSConfiguration: {
    CORSRules: [
      {
        AllowedHeaders: ["*"],
        AllowedMethods: ["GET", "PUT", "POST"],
        AllowedOrigins: ["http://localhost:5173"],
        MaxAgeSeconds: 3000,
      },
    ],
  },
});

await client.send(command);

File: src/frontend-web/scripts/configure-cors.py
import boto3
from botocore.config import Config

s3 = boto3.client(
    "s3",
    endpoint_url="http://garage:3900",
    aws_access_key_id="GKafc12d36554df7f8d5c4b7ba",
    aws_secret_access_key="abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789",
    region_name="garage",
    config=Config(s3={"addressing_style": "path"}),
)

s3.put_bucket_cors(
    Bucket="files",
    CORSConfiguration={
        "CORSRules": [
            {
                "AllowedOrigins": ["http://localhost:5173"],
                "AllowedMethods": ["GET", "PUT", "POST", "DELETE"],
                "AllowedHeaders": ["*"],
                "ExposeHeaders": ["ETag"],
                "MaxAgeSeconds": 3000,
            }
        ]
    },
)

print("✅ CORS configured on bucket files")
